core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java

```diff
@@ -394,8 +394,9 @@ public Cluster prepareCluster(Collection<HostGroup> hostGroups, Blueprint bluepr
         return cluster;
     }
 
-    public Cluster getCluster(Stack stack) {
-        return getCluster(stack.getCluster().getId());
+    public Cluster getClusterByStackResourceCrn(String stackResourceCrn) {
+        return repository.findByStackResourceCrn(stackResourceCrn).orElseThrow(NotFoundException.notFound("Cluster by stack resource crn",
```

Comment:
Nits:
* `NotFoundException.notFound` is already statically imported, so the qualified reference can be simplified to `notFound()`.
* `crn` -> `CRN`